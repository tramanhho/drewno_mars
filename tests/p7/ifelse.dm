// main : () void {
//     if (true) {
//         give "hi";
//     } else {
//         give "bye";
//     }
// }

main : () int {
	a : bool = true;
	a = 24Kmagic;
    if (a) {
        give 1;
    } else {
        give 0;
    }
    return 0;
}