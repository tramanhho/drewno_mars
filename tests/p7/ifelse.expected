.globl main

.text
main:	pushq %rbp
	movq %rsp,	%rbp
	addq $16,	%rbp
	subq $0,	%rsp

	callq magic
	movq %rdi,	%rax
	movq %rax,	-24(%rbp)
	movb -24(%rbp),	%al
	cmpb $0,	%al
	je		lbl_2
	movq $1,	%rdi
	callq printInt
	jmp		lbl_1

lbl_2:	nop
	movq $0,	%rdi
	callq printInt

lbl_1:	nop

	jmp		lbl_0
lbl_0:	addq $0,	%rsp
	popq %rbp
	retq

	movq $60,	%rax
	movq $1,	%rdi
	
	syscall
